mapie/quantile_regression.py

```diff
@@ -0,0 +1,323 @@
+                    "to a quantile method."
+                )
+        return estimator
+
+    def _check_size_calib_set(self, y, alpha) -> None:
+        value = 2/((len(y)/2)+1)
+        if alpha < value:
+            raise ValueError(
+                "The calibration set is too small."
+            )
+
+    def _check_validity_results(
+        self,
+        y_preds: NDArray,
+        y_pred_low: ArrayLike,
+        y_pred_up: ArrayLike
+    ) -> None:
+        """_summary_
+
+        Parameters
+        ----------
+        y_preds : _type_
+            _description_
+        y_pred_low : _type_
+            _description_
+        y_pred_up : _type_
+            _description_
+
+        Raises
+        ------
+        Warning
+            If the aggregated predictions of any training sample would be nan.
+        Examples
+        --------
+        >>> import warnings
+        >>> warnings.filterwarnings("error")
+        >>> import numpy as np
+        >>> from mapie.quantile_regression import _check_validity_results
+        >>> y_preds = np.array([[1, 2, 3],[3, 4, 5],[2, 3, 4]])
+        >>> y_pred_low = np.array([4, 3, 2])
+        >>> y_pred_up = np.array([4, 4, 4])
+        >>> try:
+        ...     _check_validity_results(y_preds, y_pred_low, y_pred_up)
+        ... except Exception as exception:
+        ...     print(exception)
+        ...
+        WARNING: Not correct.
+        """
+        if (
+            (y_pred_low >= y_pred_up).all() or
+                (y_preds[2] <= y_pred_low).all() or
+                (y_preds[2] >= y_pred_up).all()):
+            warnings.warn(
+                "WARNING: Not correct."
+            )
+
+    def _check_cv(
+        self,
+        cv: Optional[Union[int, str, BaseCrossValidator]] = None
+    ) -> Union[str, BaseCrossValidator]:
+        if cv is None:
+            cv == "split"
+            return cv
+        if cv == "split":
+            return cv
+        else:
+            raise ValueError(
+                "Invalid cv method."
+            )
+
+    def fit(
+        self,
+        X: ArrayLike,
+        y: ArrayLike,
+        sample_weight: Optional[ArrayLike] = None,
+    ) -> MapieQuantileRegressor:
+        # Checks
+        self._check_parameters()
+        cv = self._check_cv(self.cv)
+        estimator = self._check_estimator(self.estimator)
+        alpha = self._check_alpha(self.alpha)
+        self._check_size_calib_set(y, self.alpha)
+        alpha_copy = alpha.copy()
+        alpha_copy.append(0.5)
+        # agg_function = self._check_agg_function(self.agg_function)
+        X, y = indexable(X, y)
+        y = _check_y(y)
+        self.n_features_in_ = check_n_features_in(X, cv, estimator)
+        sample_weight, X, y = check_null_weight(sample_weight, X, y)
+        y = cast(NDArray, y)
+        # n_samples = _num_samples(y)
+
+        # Initialization
+        self.estimators_: List[RegressorMixin] = []
+
+        # Work
+        if sample_weight is None:
```

Comment:
In order to simplify the code in MAPIE and to allow the user to control the train / calib split, I would modify the strategy here :
- the user has to split the training data into train / calib sets by himself before calling MAPIE
- the user gives `X_calib`, and `y_calib` in addition to `X_train`, `y_train` in `mapie.fit()`